common/app/rendering/core/JavascriptRendering.scala

```diff
@@ -18,14 +18,9 @@ trait JavascriptRendering extends Logging {
   private implicit val scriptContext = createContext()
   private val memoizedJs: Try[EvalResult] = loadJavascript()
 
-  private def getCommonProps(props: Option[JsValue] = None): Option[JsValue] = {
-    val bundleUrl = conf.Static("javascripts/ui.bundle.browser.js")
-    val polyfillioUrl =
-      if (conf.switches.Switches.PolyfillIO.isSwitchedOn) common.Assets.js.polyfillioUrl
-      else conf.Static("javascripts/vendor/polyfillio.fallback.js")
-    val commonProps = Json.obj("bundleUrl" -> bundleUrl, "polyfillioUrl" -> polyfillioUrl)
-
-    props.map(_.as[JsObject] ++ commonProps)
+  private def getCommonProps(props: Option[JsValue] = None): Option[JsValue] = props match {
+    case None => Some(JavascriptProps.default().asJsValue().as[JsObject])
+    case _ => props.map(_.as[JsObject] ++ JavascriptProps.default().asJsValue().as[JsObject])
```

Comment:
@TBonnin what do you think of this pattern matching _par excellence_?

Reply:
http://blog.originate.com/blog/2014/06/15/idiomatic-scala-your-options-do-not-match/ 😝 

I also think we should modify this method return type. It is currently returning an `Option` but we actually always return some json

Reply:
@TBonnin Sorry, I'm not sure what you're suggesting with the link, and I'm also unsure how I would get `props.map(...)` to always return JSON rather than an option. Would it be possible to pair on this this afternoon?

Reply:
```scala
if (props.isDefined) {
  props.map(_.as[JsObject] ++ JavascriptProps.default().asJsValue().as[JsObject]).getOrElse(Json.obj)
} else {
  JavascriptProps.default().asJsValue().as[JsObject]
}
```

???

Reply:
No pb. Let's pair this afternoon. :)